src/main/java/com/iridium/iridiumskyblock/commands/JoinCommand.java

```diff
@@ -64,6 +65,12 @@ public boolean execute(CommandSender sender, String[] args) {
                         return false;
                     }
 
+                    // PR: added UserJoinEvent
+                    Optional<User> inviter = islandInvite.map(IslandInvite::getInviter);
+                    UserJoinEvent userJoinEvent = new UserJoinEvent(island.get(), user, inviter.orElse(null));
```

Comment:
Why the inviter is not offlinePlayerUser ? 
Also inviter how can be null?  Our invite command always has a inviter

Reply:
Also can you change the offlinePlayer and offlinePlayerUser to targetPlayer and targetUser

Reply:
> Also inviter how can be null? Our IslandInvite always has a inviter you can check on InviteCommand
This command also can be executed when someone staff joins to island, or not? I mean that he can do it without any invitation.

Reply:
> Why the inviter is not offlinePlayerUser ?

Because `offlinePlayerUser` suggested by you don't describe what is that variable

Reply:
Um, well you right, may we can add a boolean like isForceJoin to joincommand and join event
What your thinks about this, good or weird ?

Reply:
> Um, well you right, may we can add a boolean like isForceJoin to joincommand and join event
What your thinks about this, good or weird ?

Yes, we can add that as just a `inviter == null` check-method `isForceJoin` in the event class.
It's a current event logic i think.